sktime/forecasting/trend/_curve_fit_forecaster.py

```diff
@@ -0,0 +1,177 @@
+# !/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements FunctionForecaster."""
+
+__author__ = ["benheid"]
+__all__ = ["CurveFitForecaster"]
+
+import numpy as np
+import pandas as pd
+from scipy.optimize import curve_fit
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+
+class CurveFitForecaster(BaseForecaster):
+    """The CurveFitForecaster takes a function and tits it by using scipy curve_fit.
+
+    The CurveFitForecaster applies the scipy curve_fit method to find the best
+    parameter of a function. This function maps a list of integers to real values.
+    The integers are derrived from the index of the input time series.
+
+    In `fit`
+    1. The index of the input time series is transformed to a list of Integers.
```

Comment:
so, it discards the index and uses the `iloc` (not `loc`) index?

Reply:
Basically yes. However, I extract it without using iloc, since this probably would cause problems if the forecasting horizon is not continuously.

Reply:
can you add some explanation about the "How"?